app/backend/main.py

```diff
@@ -1,3 +1,12 @@
+import os
+
 from app import create_app
+from load_azd_env import load_azd_env
+
+# WEBSITE_HOSTNAME is always set by App Service, RUNNING_IN_PRODUCTION is set in main.bicep
+RUNNING_ON_AZURE = os.getenv("WEBSITE_HOSTNAME") is not None or os.getenv("RUNNING_IN_PRODUCTION") is not None
```

Comment:
This code also exists in two places- 
1) here, so that we load the environment when starting up the app locally
2) in app.py, so that we can decide what kind of credential to use